jupyterhub_ssh/__init__.py

```diff
@@ -221,17 +222,37 @@ class JupyterHubSSH(Application):
         config=True,
     )
 
-    # FIXME: Make this a yarl.URL object?
     hub_url = Any(
         "",
         help="""
-        URL of JupyterHub to connect to.
+        URL of JupyterHub's proxy to connect to.
+        
+        `jupyterhub-ssh` needs to be able to connect to both the
+        JupyterHub API and to the user's server. If running in
+        the same cluster as the JupyterHub, this should point to
+        the `proxy-public` service. When running outside the same
+        cluster, this should be set to the user accessible URL of
+        the hub.
 
         *Must* be set.
+        
+        Examples:
+        
+        `http://proxy-public` (in same cluster & namespace as jupyterhub)
+        `https://my-hub-url.com` (when running outside the cluster)
```

Comment:
```suggestion
        jupyterhub-ssh needs to be able to connect to both the JupyterHub API
        (/hub/api) and to the users' servers (/user/<username>) via HTTP or
        HTTPS. This URL doesn't have to be the public URL though as the URL is
        only be used by jupyterhub-ssh itself.

        *Must* be set.
        
        Examples:
        
        - If jupyterhub-ssh is deployed in the same Kubernetes cluster and
          namespace as the official JupyterHub Helm chart, you can use
          `http://proxy-public:8000` or `http://proxy-http:8000` depending on
          how the JupyterHub Helm chart is configured. Use
          `http://proxy-http:8000` if the JupyterHub Helm chart has been
          configured with `proxy.https.enabled=true` and
          `proxy.https.type=letsencrypt`.

        - If jupyterhub-ssh can't access the JupyterHub via local network or you
          don't trust the local network to be secure, use a public URL with
          HTTPS such as `https://my-hub-url.com`.
```

Reply:
@yuvipanda tweaked the description to not assume jupyterhub-ssh ran inside a k8s cluster, added port `:8000`, and clarified that it may be `proxy-http` rather than `proxy-public`